js/reducers/report-reducer.ts

```diff
@@ -253,7 +253,7 @@ export default function report(state = new ReportState({}), action?: any) {
       return processReportItemRequests(state);
     case SET_REPORT_ITEM_ANSWER:
       const answer = getAnswer(state, action.questionId, action.reportItemAnswer.platformUserId);
-      const storageName = action.reportItemAnswer.itemsType === "fullAnswer" ? "reportItemAnswersFull" : "reportItemAnswersCompact";
+      const storageName = action.reportItemAnswer.itemsType === "compactAnswer" ? "reportItemAnswersCompact" : "reportItemAnswersFull";
```

Comment:
@emcelroy,

> In the report reducer, storageName was incorrectly being set to "reportItemAnswersCompact" when itemsType was undefined. As I understand it, storageName should be set to "reportItemAnswersFull" in that situation.

This change fixes the problem you described, but I'd say it's pretty hidden. Initially, I thought nothing has changed here. :) I can imagine it might be randomly changed by another developer not aware of this logic.

I'd vote to change it to a more explicit statement with `if` where we check for undefined value first. And/or some comment could be useful.

Reply:
@pjanik Are the latest changes to report-reducer.ts what you were thinking?

I couldn't think of a way to make it more concise, but maybe that's OK?

Reply:
My point is that I'd personally make it less concise. To handle `undefined` value in a very explicit way, so this logic is not broken accidentally later. For example:

```javascript
let storageName;
if (action.reportItemAnswer.itemsType === "compactAnswer") {
  storageName = "reportItemAnswersCompact";
} else if (action.reportItemAnswer.itemsType === "fullAnswer") {
  storageName = "reportItemAnswersFull";  
} else { 
  // Old interactives might not send back itemsType. Storage name should default to "reportItemAnswersFull".
  storageName = "reportItemAnswersFull"
}
```

It might be personal, so ofc it's up to you. I just thought that `? :` construct makes the handling of `undefined` value a bit hidden.

Reply:
@pjanik Thanks. I had actually tried something very similar to that first, but was getting an error: `Type 'string' is not assignable to type 'keyof IReportState'.` Looks like if we cast `storageName` to `keyof IReportState` that error goes away. See latest changes. Does that seem OK to you?

Reply:
One note is that based on the code coverage, it seems like there are no tests of the `reportItemAnswersCompact`. I'm a little surprised by that. But I guess it means the cypress tests do not cover this case of a compact answer.

I don't think adding that test needs to be part of this PR.